azuredevops.go

```diff
@@ -106,6 +112,20 @@ func (m importModule) runAzureDevOpsHandler(c *gin.Context) {
 	c.Status(http.StatusCreated)
 }
 
+func parseRepoRefParams(wharfGroupName, wharfProjectName string) (azureOrgName, azureProjectName, azureRepoName string) {
+	slashIndex := strings.IndexRune(wharfGroupName, '/')
+	if slashIndex == -1 {
```

Comment:
The 1 here is a magic number- refactor to named const for better readability.

Reply:
Ex. `noSlashInString`

Reply:
I'm torn about this one. I am probably bias but it's common convention throughout most programming languages to have `IndexOf` functions that return -1 when not found.

But I extracted the lower-level string splitting part to a different function where a const like `noSlashInString`/`notFoundIndex` doesn't bloat the block.

I find the following having better readability:

```go
func parseRepoRefParams(wharfGroupName, wharfProjectName string) (azureOrgName, azureProjectName, azureRepoName string) {
	azureOrgName, azureProjectName = splitStringOnceRune(wharfGroupName, '/')
	if azureProjectName == "" {
		azureProjectName = wharfProjectName
	}
	azureRepoName = wharfProjectName
	return
}

func splitStringOnceRune(value string, delimiter rune) (a, b string) {
	const notFoundIndex = -1
	delimiterIndex := strings.IndexRune(value, delimiter)
	if delimiterIndex == notFoundIndex {
		a = value
		b = ""
		return
	}
	a = value[:delimiterIndex]
	b = value[delimiterIndex+1:] // +1 to skip the delimiter
	return
}
```

Using a comment for the +1 part as I found that extracting that to a variable bloated the code again.

Reply:
I agree about -1 being such a well established return value from "indexOf" variants when not found, so explaining that is a bit like doing this:

```cs
i++; // Increment the i variable by 1
```